openedx/core/djangoapps/content_libraries/management/commands/migrate_lib_to_learning_core.py

```diff
@@ -0,0 +1,234 @@
+                for file_path, metadata in draft_files.items()
+                if file_path.endswith("definition.xml")
+            }
+            for file_path, draft_metadata in draft_definition_files.items():
+                published_metadata = published_metadata_dict.get(file_path)
+                if draft_metadata.modified:
+                    block_type, block_id, _def_xml = file_path.split('/')
+                    xml_bytes = get_bundle_file_data(bundle.uuid, file_path, use_draft=lib_constants.DRAFT_NAME)
+                    display_name = extract_display_name(xml_bytes, file_path)
+
+                    # If this is newly created in the draft, we have to create a
+                    # whole new Component...
+                    if published_metadata is None:
+                        component = components_api.create_component(
+                            learning_package_id=learning_package.id,
+                            namespace=COMPONENT_NAMESPACE,
+                            type=block_type,
+                            local_key=block_id,
+                            created=now,
+                            created_by=None,
+                        )
+                        component_pk = component.pk
+                        version_num = 1
+                    # Otherwise, it's just been modified...
+                    else:
+                        component_pk = published_component_pks[file_path]
+                        version_num = 2
+
+                    component_version = components_api.create_component_version(
+                        component_pk=component_pk,
+                        version_num=version_num,
+                        title=display_name,
+                        created=now,
+                        created_by=None,
+                    )
+                    text_content, _created = contents_api.get_or_create_text_content_from_bytes(
+                        learning_package_id=learning_package.id,
+                        data_bytes=xml_bytes,
+                        mime_type=f"application/vnd.openedx.xblock.v1.{block_type}+xml",
+                        created=now,
+                    )
+                    components_api.add_content_to_component_version(
+                        component_version,
+                        raw_content_id=text_content.pk,
+                        key="definition.xml",
+                        learner_downloadable=False
+                    )
+
+            # Now remove stuff that was present in the published set but was
+            # deleted in the draft.
+            deleted_definition_files = set(published_definition_files) - set(draft_definition_files)
+            for deleted_definition_file in deleted_definition_files:
+                log.info(f"Deleting {deleted_definition_file} from draft")
+                component_pk = published_component_pks[deleted_definition_file]
+                publishing_api.set_draft_version(component_pk, None)
```

Comment:
(openedx-learning API feedback, not blocking for this PR)

I'm enjoying this PR because it's cool seeing the publishing and components API in action in edx-platform. For the most part, each call to the APIs has been very clear and self-explanatory. Kudos.

For this particular `set_draft_version` call, though, I had to dive into [the definition](https://github.com/openedx/openedx-learning/blob/v0.3.6/openedx_learning/core/publishing/api.py#L203-L213) to understand that passing in None means "remove from the draft". It feels like that function does one of two different things depending on whether the second arg is None.  I wonder if, rather than having a single API function which takes an Optional, it'd be better to have one API function which demands and int, and another which removes the draft:
```python
def set_draft_version(publishable_entity_id: int, publishable_entity_version_pk: int) -> None:
    ...

def remove_from_draft(publishable_entity_id: int) -> None:
    ...
```

Reply:
Good point. I suspect I was just getting too wrapped up in the data model to step back and think about it from the API consumer's point of view. I'll do this.